### ISSUE TITLE
Remove MRCCompatClient

### ISSUE THREAD

**Issue by jaellio** [openservicemesh/osm#5040]
Part of #4998 

The MRCCompatClient was created to support the transition to using a MeshRootCertificate custom resource to configure the certificate provider on osm install and facilitate OSM root certificate rotation. As specified in #4718, the MRCCompatClient allowed us to support a single implementation of certificate.Manager regardless of whether or not an MRC resource was used to configure the certificate provider. If the `enableMeshRootCertificate` binary flag was set to false an MRCCompatClient was created.  

In v1.3 a MeshRootCertificate resource will be created on install by default and the MRCCompatClient is no longer needed. This issue tracks the removal of the MRCCompatClient.

Resources:
- MRCCompatClient added in #4718 
-  #4816 
- `enableMeshRootCertificate` flag added in #4871 

The removal of the `enableMeshRootCertificate` flag is tracked in #5041.

**Comment by jaellio** [comment]
Blocked on:
- #4818 
- #4723 
- #4815 
- #4962

**Comment by ceekay47** [comment]
Hi @jaellio , I would to like work on this. Can you please help me get started? Thanks!

**Comment by jaellio** [comment]
HI @ceekay47! Thank you so much for your interest in picking up this issue! This change is currently blocked on several other issues. Let me see if there is another good first issue that is currently unblocked.

**Comment by ceekay47** [comment]
Sure @jaellio. That would be really awesome. Thanks!
